### PR TITLE
[move] Add error messages for linker error #9999

### PR DESCRIPTION
**Pull request by aalok-t**
### Description

Add error messages for `LINKER_ERROR`, fixing https://github.com/aptos-labs/aptos-core/issues/9999 (part of https://github.com/aptos-labs/aptos-core/issues/9580). 

This includes error messages for 5 tests and 1 for Move VM.

### CODE REVIEW

**Review by vineethk — CHANGES_REQUESTED** [review]
Tests are failing:

```
error[E0599]: no method named `with_message` found for enum `StatusCode` in the current scope
   --> third_party/move/move-vm/runtime/src/unit_tests/vm_arguments_tests.rs:634:53
    |
634 |         assert_eq!(status, StatusCode::LINKER_ERROR.with_message(format!("Linker Error: Signature {:?} is deprecated", signature)));
    |                                                     ^^^^^^^^^^^^ method not found in `StatusCode`

error[E0599]: no method named `with_message` found for enum `StatusCode` in the current scope
   --> third_party/move/move-vm/runtime/src/unit_tests/vm_arguments_tests.rs:817:63
    |
817 |     assert_eq!(error.major_status(), StatusCode::LINKER_ERROR.with_message(format!("Linker Error: Call to item at a non-existent external module {:?}", module)));
    |                                                               ^^^^^^^^^^^^ method not found in `StatusCode`

For more information about this error, try `rustc --explain E0599`.
```

**Review by brmataptos — CHANGES_REQUESTED** [review]

File: aptos-move/e2e-testsuite/src/tests/scripts.rs [hunk 1]
```diff
@@ -85,7 +85,8 @@ fn script_none_existing_module_dep() {
     script
         .address_identifiers
         .push(AccountAddress::new([2u8; AccountAddress::LENGTH]));
-    script.identifiers.push(Identifier::new("module").unwrap());
+    let module = Identifier::new("module").unwrap();
```
[the remark this answers is not in the document]
Comment (aalok-t) [reply aptos-move/e2e-testsuite/src/tests/scripts.rs:88]: Yes, but that didn't work well so I decided against it. I will remove this line. Thanks Brain!

**Review by brmataptos — APPROVED** [review]

File: aptos-move/e2e-testsuite/src/tests/scripts.rs [hunk 2]
```diff
@@ -130,7 +130,8 @@ fn script_none_existing_module_dep() {
         //StatusCode::LINKER_ERROR
```
Comment (vineethk) [inline aptos-move/e2e-testsuite/src/tests/scripts.rs:130]: Remove comment, no longer useful.

File: third_party/move/move-binary-format/src/file_format.rs [hunk 3]
```diff
@@ -384,7 +384,7 @@ impl StructDefinition {
         match &self.field_information {
             // TODO we might want a more informative error here
```
Comment (vineethk) [inline third_party/move/move-binary-format/src/file_format.rs:385]: Perhaps this change addresses this TODO, so remove comment?

**Review by vineethk — COMMENTED** [review]
Minor comments to address. lgtm otherwise.

**Review by vineethk — APPROVED** [review]

**Pull request merged**